static/app/components/smartSearchBar/index.tsx

```diff
@@ -384,10 +389,25 @@ class SmartSearchBar extends React.Component<Props, State> {
   }
 
   async doSearch() {
-    const {onSearch, onSavedRecentSearch, api, organization, savedSearchType} =
-      this.props;
     this.blur();
+
+    if (!this.hasValidSearch && this.hasImprovedSearch) {
+      return;
+    }
+
     const query = removeSpace(this.state.query);
+    const {onSearch, onSavedRecentSearch, api, organization, savedSearchType} =
+      this.props;
+
+    trackAnalyticsEvent({
```

Comment:
I moved this up so that we can properly record analytics when the search is triggered outside of `onSubmit`